lib/puppet/provider/grub_config/grub2.rb

```diff
@@ -101,6 +101,9 @@ def flush
     fail("Cannot find grub.cfg location to use with grub-mkconfig") unless cfg
 
     super
-    mkconfig "-o", cfg
+
+    if mkconfig
+      mkconfig "-o", cfg
+    end
```

Comment:
Parameters are retrieved with `resource[:param]`, so

```suggestion

    mkconfig "-o", cfg if resource[:run_mkconfig]
```

should do